packages/wdio-sauce-service/src/service.ts

```diff
@@ -1,13 +1,79 @@
-import SauceLabs from 'saucelabs'
+import SauceLabs, { Job, SauceConnectOptions, SauceLabsOptions } from 'saucelabs'
 import logger from '@wdio/logger'
 import { isUnifiedPlatform } from './utils'
+import WDIOReporter from '@wdio/reporter'
+import { DesiredCapabilities } from 'webdriver'
+import { BrowserObject, HookFunctions, Suite } from 'webdriverio'
+import { CucumberHookFunctions, CucumberHookObject, CucumberHookResult } from '@wdio/cucumber-framework'
+import { ScenarioResult } from 'cucumber'
 
 const jobDataProperties = ['name', 'tags', 'public', 'build', 'custom-data']
 
 const log = logger('@wdio/sauce-service')
 
-export default class SauceService {
-    constructor (options) {
+declare global {
+    namespace NodeJS {
+        interface Global {
+            browser: WebdriverIO.BrowserObject;
+        }
+    }
+}
+
+declare module 'webdriverio' {
+    export interface Test extends Suite {
+        // Jasmine only
+        description?: string
+        // Mocha only
+        retriedTest?: any
+        retries: number
+        currentRetry: number
+    }
+}
+
+export interface SauceServiceConfig {
+    /**
+     * If true it runs Sauce Connect and opens a secure connection between a Sauce Labs virtual
+     * machine running your browser tests.
+     */
+    sauceConnect?: boolean;
+    /**
+     * Apply Sauce Connect options (e.g. to change port number or logFile settings). See this
+     * list for more information: https://github.com/bermi/sauce-connect-launcher#advanced-usage
+     */
+    sauceConnectOpts?: SauceConnectOptions;
+    /**
+     * Use Sauce Connect as a Selenium Relay. See more [here](https://wiki.saucelabs.com/display/DOCS/Using+the+Selenium+Relay+with+Sauce+Connect+Proxy).
+     * @deprecated
+     */
+    scRelay?: boolean;
+    /**
+     * If true it updates the job name at the Sauce Labs job in the beforeSuite Hook.
+     * Attention: this comes at the cost of an additional call to Sauce Labs
+     * Default: false
+     */
+    setJobNameInBeforeSuite?: boolean;
+}
+
+export type SauceConfig = SauceLabsOptions & WebdriverIO.Config;
+
+interface MultiRemoteBrowser extends WebdriverIO.BrowserObject {
+    instances: string[]
+    isMultiremote: boolean
+}
```

Comment:
do they have the same kind of type annotation? If so, we could think about extracting it somewhere more global, but anyways we could always move this in a later PR into a different package when there is the need for it.

Reply:
Yes I agree it could be a separate issue. But we should think of some solutions now, nonetheless, or our repos will litter with duplicate types like this. I saw some similar problems when you want to cross reference types from other packages such as mocha framework, which extends Webdriverio types.

Reply:
I personally like not to prematurely optimize things - my preferred approach is usually to adapt things to be as generic as they need to be, once it's needed. So I would rather start with having this `interface` here (as it's the only place it's used right now) and refactoring it to a different place as soon as the need for it arises. And where to put it then will be something to look at specifically in the context. Especially considering we are not exporting this type, so we don't run risk of any breaking changes when we move this to a different place.